JabRef/jabref
# Fix for issue 6877: Allow users to customize the API Key

Fixes #6877 



- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

Changed:
1. Add feature: allows the user to customize the Springer API Key in the customization tab of the preferences and check this key.
2. Update API_URL's protocol of SpringerFetcher to HTTPS.

<br>

Initial default value, do not use custom API key:

![preferences-customization-tab](https://user-images.githubusercontent.com/47767371/117575346-f2be8800-b113-11eb-9062-0f50dbe9100a.png)

Check success with a valid API key:

![successful](https://user-images.githubusercontent.com/47767371/117575350-f7833c00-b113-11eb-84ea-0cc37949e002.png)

Check failed with an invalid API key:

![failed](https://user-images.githubusercontent.com/47767371/117575354-fa7e2c80-b113-11eb-860e-29b5ae42b2ef.png)

Search using a custom Springer API key:

![search-lidar](https://user-images.githubusercontent.com/47767371/117575358-fe11b380-b113-11eb-9791-be03f315e2be.png)
